lib/administrate/field/select_basic.rb

```diff
@@ -8,23 +8,33 @@ class Engine < ::Rails::Engine
       end      
       
       def choices
-        options.fetch(:choices, []).
-          map { |o| prettify? ? [prettify(o), o] : o }
+        options.fetch(:choices, []).map { |o| convert(o) }
       end
   
       def to_s
-        prettify? ? prettify(data) : data
+        convert(data, raw: true)
       end
   
       def include_blank
         options.fetch(:include_blank, false)
       end
     
       private
+
+      def convert item, raw: false
+        return [prettify(item), item] if prettify? && !raw
+        return prettify(item) if prettify? && raw
+        return internationalise(item) if i18n?
+        return item
+      end
```

Comment:
I think it's OK - might be better to break some of those double-conditions into separate conditions but it's fine for now.

Quick question: if I'm not mistaken, doesn't the named argument make this only work in ruby 2.3+ ? I think administrate is supported by earlier versions so might it be worth using some old-school syntax here?